core/backend/src/ViewStore.ts

```diff
@@ -0,0 +1,1397 @@
+ * View names and ViewGroup names may not contain either "/" or "@". ViewGroups are stored in the "viewGroups" table.
+ *
+ * Views may be "tagged" with one or more Tags. Tags are named with an arbitrary string that can be used to group Views. A Tag may
+ * be associated with multiple Views, and a View may have multiple Tags. Tags are stored in the "tags" table.
+ *
+ * Views may optionally have a thumbnail, paired via the View's Id. Thumbnails are stored in the "thumbnails" table.
+ *
+ * Note: All ElementIds and ModelIds in ModelSelectors, CategorySelectors, DisplayStyles, Timelines, etc. are converted to guid-based identifiers when stored in the ViewStore.
+ * They are then remapped back to their Ids when loaded from the ViewStore. This allows the ViewStore to be used with more than one iModel,
+ * provided that the same Guids are used in each iModel. This is done by storing the set of unique Guids in the "guids" table, and then
+ * creating a reference to the row in the "guids" table via the special Id prefix "^". For example, if a category selector contains the
+ * Id "0x123", then the guid from element 0x123 is stored in the "guids" table, and the category selector is stored with the rowId of the entry
+ * in the guid table (e.g. "^1w"). When the category selector is loaded from the ViewStore, the guid is looked up in the "guids" table and
+ * the iModel is queried for the element with that guid. That element's Id (which may or may not be 0x123) is then returned in the category selector.
+ *
+ * @beta
+ */
+export namespace ViewStore {
+
+  export const tableName = {
+    categorySelectors: "categorySelectors",
+    displayStyles: "displayStyles",
+    viewGroups: "viewGroups",
+    guids: "guids",
+    modelSelectors: "modelSelectors",
+    taggedViews: "taggedViews",
+    tags: "tags",
+    thumbnails: "thumbnails",
+    timelines: "timelines",
+    searches: "searches",
+    views: "views",
+  } as const;
+
+  /** data for a Thumbnail */
+  type ThumbnailData = Uint8Array;
+
+  /** A row in a table. 0 means "not present" */
+  export type RowId = number;
+
+  /** a string representation of a row in a table of a ViewStore. Will be a base-36 integer with a leading "@" (e.g."@4e3") */
+  type RowString = string;
+
+  /** a string representation of a row in the Guid table. Will be a base-36 integer with a leading "^" (e.g."^4e3") */
+  type GuidRowString = string;
+
+  /** common properties for all tables */
+  interface TableRow {
+    name?: string;
+    json: string;
+    owner?: string;
+  }
+  type DisplayStyleRow = TableRow;
+  type SelectorRow = TableRow;
+  type TagRow = TableRow;
+  type SearchRow = TableRow;
+  type TimelineRow = TableRow;
+
+  /** a row in the "views" table */
+  interface ViewRow extends MarkRequired<TableRow, "name"> {
+    className: string;
+    groupId: RowId;
+    isPrivate?: boolean;
+    modelSel?: RowId;
+    categorySel: RowId;
+    displayStyle: RowId;
```

Comment:
Since model selector, category selector and display styles are separate from a view, what will be the procedure to update thumbnails of all affected views when an update occurs?

Reply:
Thumbnails are only approximations and can only be updated from a view in the frontend. They can be out-of-date and still be useful. There's no expectation that they'll be updated other than one-by-one manually.